modules/javafx.graphics/src/main/java/javafx/scene/Node.java

```diff
@@ -1404,6 +1405,87 @@ public String getName() {
         return visible;
     }
 
+    /**
+     * Whether or not this {@code Node} is showing (that is, it is part of an
+     * open Window on the user's system). The Window might be "showing", yet the user might not
+     * be able to see it due to the Window being rendered behind another Window
+     * or due to the Window being positioned off the monitor.
+     *
+     * <p>Note that the {@code Node} does not need to be visible for this property
+     * to be {@code true}.
+     *
+     * @defaultValue false
+     * @since 20
+     */
+    private ReadOnlyBooleanProperty shown;
+
+    public final boolean isShown() {
+        Scene s = getScene();
+        if (s == null) return false;
+        Window w = s.getWindow();
+        return w != null && w.isShowing();
```

Comment:
Are you avoiding calling `get` on the property to avoid its initialization?

Reply:
I would suggest adding a comment saying that this is done to avoid initialization.

I'm not sure that it's that critical for performance, to be honest.

Reply:
I think the pattern is good to do, the `shownProperty` will only be used for a small number of nodes, and not initializing it when not needed is a common pattern in `Node` itself already.  Here for example for `opacity`:

    private DoubleProperty opacity;

    public final void setOpacity(double value) {
        opacityProperty().set(value);
    }
    public final double getOpacity() {
        return opacity == null ? 1 : opacity.get();
    }

But it is far from the only one.  `disabled` does it, `pickOnBounds`, `blendMode`, etc.

None of these contain a comment as to why, but I'll add one for `shown`.